noir_stdlib/src/meta/typ.nr

```diff
@@ -3,34 +3,54 @@ use crate::option::Option;
 
 impl Type {
     #[builtin(type_as_array)]
+// docs:start:as_array
     fn as_array(self) -> Option<(Type, Type)> {}
+    // docs:end:as_array
 
     #[builtin(type_as_constant)]
+// docs:start:as_constant
     fn as_constant(self) -> Option<u32> {}
+    // docs:end:as_constant
 
     #[builtin(type_as_integer)]
+// docs:start:as_integer
     fn as_integer(self) -> Option<(bool, u8)> {}
+    // docs:end:as_integer
 
     #[builtin(type_as_slice)]
+// docs:start:as_slice
     fn as_slice(self) -> Option<Type> {}
+    // docs:end:as_slice
 
     #[builtin(type_as_struct)]
+// docs:start:as_struct
     fn as_struct(self) -> Option<(StructDefinition, [Type])> {}
+    // docs:end:as_struct
 
     #[builtin(type_as_tuple)]
+// docs:start:as_tuple
     fn as_tuple(self) -> Option<[Type]> {}
+    // docs:end:as_tuple
 
     #[builtin(type_get_trait_impl)]
+// docs:start:get_trait_impl
```

Comment:
some of this formatting is a little funky

Reply:
Yeah. I used to have it aligned differently but nargo fmt informed me I was wrong.